backend/dataall/core/environment/api/resolvers.py

```diff
@@ -71,11 +72,25 @@ def check_environment(context: Context, source, account_id, region):
                 action='CHECK_PIVOT_ROLE',
                 message='Pivot Role has not been created in the Environment AWS Account',
             )
+    mlStudioEnabled = None
+    for parameter in data.get("parameters", []):
+        if parameter['key'] == 'mlStudiosEnabled':
+            mlStudioEnabled = parameter['value']
+
+    if mlStudioEnabled and data.get("mlStudioVPCId", None) and data.get("mlStudioSubnetIds", []):
+        log.info("Check if ML Studio VPC Exists in the Account")
+        EC2.check_vpc_exists(
+            AwsAccountId=account_id,
+            region=region,
+            role=cdk_look_up_role_arn,
+            vpc_id=data.get("mlStudioVPCId", None),
+            subnet_ids=data.get('mlStudioSubnetIds', []),
+        )
```

Comment:
If we are very purist with the modularization principles, this snippet of code should be defined in the `mlstudio` module. We have 2 options: either we check the VPCid and subnets `if data.get("mlStudioVPCId", None) and data.get("mlStudioSubnetIds", []):` without referencing `mlStudioEnabled` or we create an abstraction

Reply:
Looking into the PR a bit more, we do not even need this check (I think), because it is duplicated in `dataall.modules.mlstudio.services.mlstudio_service.SagemakerStudioService.check_mlstudio_domain_vpc`

Reply:
The reason the check is both in ML Studio Domain creation AND in creation/update of a new environment is because of the following potential bug if the check was not in environment creation/update:

- When a user creates a new environment and specifies a VPC for their ML Studio Studio Domain
- If the VPC does not exist the environment will still get created properly but the `createSageMakerStudioDomain` API will fail
  - Meaning the environment will create and say ML Studio enabled but the environment stack will not actually create the domain and no ML Studio Users will be able to get created 
  - A user would then have to disable ML Studio and re-enable ML Studio (in 2 subsequent edits) to get the environment stack to deploy the appropriate resources (i.e. the sagemaker domain)

Reply:
Switched to `vpcId` and `subnetIds`